### DISCUSSION TITLE
node_modules/socket.io/dist/socket.d.ts:63:39 - error TS1005: ',' expected

### DISCUSSION THREAD

**Discussion by RonnieRocket147** [socketio/socket.io#4292] [General] [answered]
Hello,

I tried upgrading the server from 4.0.0 to the latest 4.4.1 but receive the following error when building the project:

```
node_modules/socket.io/dist/socket.d.ts:63:39 - error TS1005: ',' expected.

63 export declare type Event = [eventName: string, ...args: any[]];
                                         ~

node_modules/socket.io/dist/socket.d.ts:63:39 - error TS1005: ',' expected.

63 export declare type Event = [eventName: string, ...args: any[]];
                                                          ~

Found 2 errors.
```

Node version is `v16.13.0`

**Comment by darrachequesne** [comment, marked as answer]
Hi! Which version of TypeScript are you using? Labeled tuple were added in TypeScript 4.0.

Related: https://github.com/socketio/socket.io/commit/02c87a85614e217b8e7b93753f315790ae9d99f6 (not released yet)

**Reply by RonnieRocket147** [reply]
Thanks @darrachequesne ,

It seems I was using `v3.9.5` so will upgrade to `v4+`

Thanks!
